### PR TITLE
Video Compression Using Container (ECS,ECR)  - Eftakhar

### PR DESCRIPTION
**Pull request by shoccho**
Fixes #

- [ ] Does not depend on other PRs
- [x] Depends on other PR being merged first: #324 

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [x] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Added new functionality to convert the video files using a container from ECR via ECS tasks instead of lambda
## What was done
 - Added new Docker image creation files ( Docker files and shell scripts to push Image to ECR )
 - Added new ECS Task defination resource
 - Added new Task log group resource 
 - Added permissions to lambda to run ecs task and pass role
 - Added new Conversion mechanism with ecs tasks
 - Removed older video conversion mechanism code  
 - Included the terraform backend init script to initialize terraform deployment with s3 backend
 - 
## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

**Review by mhazad — CHANGES_REQUESTED** [review]
Please address the feedback.

File: converter/fileoperations.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
+aws s3api get-object --bucket $BUCKET_NAME --key $INPUT_FILE_NAME input
+
+ffmpeg -y -i input -filter_complex scale=720:480 output.mp4
+aws s3 cp output.mp4 s3://$BUCKET_NAME/$OUTPUT_FILE_NAME
```
[the remark this answers is not in the document]
Comment (shoccho) [reply converter/fileoperations.sh]: Implemented this in JavaScript in the commit [d8f9037](https://github.com/CraftsmenLtd/coe-batch-q322/pull/325/commits/d8f90376cec59f8c30900c2a5113014f9cad02ea)

File: serverless.yml [hunk 2]
```diff
@@ -203,6 +215,65 @@ resources:
...
+              Version: "2012-10-17"
+              Statement:
+                - Effect: Allow
+                  Action:
+                    - "ecr:GetAuthorizationToken"
+                    - "ecr:BatchCheckLayerAvailability"
+                    - "ecr:GetDownloadUrlForLayer"
+                    - "ecr:BatchGetImage"
+                    - "logs:*"
+                    - "ecs:*"
```
[the remark this answers is not in the document]
Comment (shoccho) [reply serverless.yml]: Removed unnecessary permissions in this commit [c76fc9d](https://github.com/CraftsmenLtd/coe-batch-q322/pull/325/commits/c76fc9d2acf027513f112a87c413b8fcef53b3d1)

**Review by mhazad — APPROVED** [review]
LGTM

**Pull request merged**